src/imgui/ImGuiSymbols.cc

```diff
@@ -189,15 +262,56 @@ void ImGuiSymbols::paint(MSXMotherBoard* /*motherBoard*/)
 
 		im::TreeNode("Symbols per file", ImGuiTreeNodeFlags_DefaultOpen, [&]{
 			auto drawFile = [&](const FileInfo& info) {
+				auto* file = symbolManager.findFile(info.filename);
 				im::StyleColor(!info.error.empty(), ImGuiCol_Text, getColor(imColor::ERROR), [&]{
 					auto title = strCat("File: ", info.filename);
 					im::TreeNode(title.c_str(), [&]{
 						if (!info.error.empty()) {
 							ImGui::TextUnformatted(info.error);
 						}
 						im::StyleColor(ImGuiCol_Text, getColor(imColor::TEXT), [&]{
+							auto arrowSize = ImGui::GetFrameHeight();
+							auto extra = arrowSize + 2.0f * style.FramePadding.x;
+							ImGui::SetNextItemWidth(ImGui::CalcTextSize("3-3").x + extra);
+						        ImGui::AlignTextToFramePadding();
+
+							auto preview = formatSlot(file->slot, motherBoard);
+				                        im::Combo(tmpStrCat("Slot##", info.filename).data(), preview.c_str(), [&]{
+								// Set slot and all the symbols in it
+								auto setSlot = [&](std::optional<uint8_t> newSlot) {
+ 									file->slot = newSlot;
+									for (auto& symbol : file->getSymbols()) {
+										symbol.slot = newSlot;
+									}
+								};
+								// initial state
+								if (ImGui::Selectable("-", !file->slot)) {
+									setSlot({});
+								}
+
+								int psSs = 0;
+								for (uint8_t ps = 0; ps < 4; ++ps) {
+									if (isSlotExpanded(motherBoard, ps)) {
+										for (uint8_t ss = 0; ss < 4; ++ss) {
+											auto slotInfo = tmpStrCat(ps, "-", ss);
+											if (ImGui::Selectable(slotInfo.c_str(), psSs == file->slot)) {
+												setSlot(psSs);
+											}
+											psSs += 1 << 2; // add to subslot bits
+										}
+										psSs++;
```

Comment:
I still believe this is wrong. The 2x2 bits that represent the ps and ss are now correct, but after the loop over ss and after this statement `psSS` has increased by 17. So bits 4-7 are no longer zero. And that will cause problems in the comparison `psSS == file->slot` because that compares all 8 bits.

You could fix that by adding an extra `& 15` in the comparison. But I think it's simpler to:
* remove lines 292, 300, 302, 307
* add this after line 296: int psSs = (ss << 2) + ps;
* in line 304 change `psSS` to `ps`